src/Umbraco.Web/Editors/LanguageController.cs

```diff
@@ -35,59 +42,119 @@ public IEnumerable<Culture> GetAllCultures()
+            }
+
+            return langs.OrderBy(x => x.Name);
         }
-        
+
+        [HttpGet]
+        public Language GetLanguage(int id)
+        {
+            var lang = Services.LocalizationService.GetLanguageById(id);
+            if (lang == null)
+                throw new HttpResponseException(Request.CreateResponse(HttpStatusCode.NotFound));
+
+            var model = Mapper.Map<Language>(lang);
+
+            //if there's only one language, by default it is the default
+            var allLangs = Services.LocalizationService.GetAllLanguages().OrderBy(x => x.Id).ToList();
+            if (!lang.IsDefaultVariantLanguage)
+            {
+                if (allLangs.Count == 1)
+                    model.IsDefaultVariantLanguage = true;
+                else if (allLangs.All(x => !x.IsDefaultVariantLanguage))
+                {
+                    //if no language has the default flag, then the defaul language is the one with the lowest id
+                    model.IsDefaultVariantLanguage = allLangs[0].Id == lang.Id;
+                    model.Mandatory = allLangs[0].Id == lang.Id;
+                }
+            }
+            
+
+            return model;
+        }
+
         /// <summary>
         /// Deletes a language with a given ID
         /// </summary>
         [HttpDelete]
         [HttpPost]
-        public void DeleteLanguage(int id)
+        public IHttpActionResult DeleteLanguage(int id)
         {
             var language = Services.LocalizationService.GetLanguageById(id);
-            if (language == null)
-            {
-                throw new EntityNotFoundException(id, $"Could not find language by id: '{id}'.");
-            }
+            if (language == null) return NotFound();
 
-            if (language.IsDefaultVariantLanguage)
+            var totalLangs = Services.LocalizationService.GetAllLanguages().Count();
+
+            if (language.IsDefaultVariantLanguage || totalLangs == 1)
             {
-                var message = $"Language with id '{id}' is currently set to 'default' and can not be deleted.";
-                throw new HttpResponseException(Request.CreateErrorResponse(HttpStatusCode.BadRequest, message));
+                var message = $"Language '{language.IsoCode}' is currently set to 'default' or it is the only installed language and can not be deleted.";
+                throw new HttpResponseException(Request.CreateNotificationValidationErrorResponse(message));
             }
 
             Services.LocalizationService.Delete(language);
+
+            return Ok();
         }
 
         /// <summary>
-        /// Saves a bulk set of languages with default/mandatory settings and returns the full set of languages configured.
+        /// Creates or saves a language
         /// </summary>
         [HttpPost]
-        public IEnumerable<Language> SaveLanguages(IEnumerable<Language> languages)
+        public Language SaveLanguage(Language language)
```

Comment:
So the signature has changed here - was there some form of UI intention to do a bulk save of languages @Shazwazza  previously that has now been scrapped?

Reply:
We are not bulk saving